texastribune/donations
# Detect secrets pre-commit

#### What's this PR do?

Adds a baseline for secret detection. Looks for new high-entropy strings and warns in case they might be keys or passwords.

#### Why are we doing this? How does it help us?

It's good not to leak secrets.

#### How should this be manually tested?

- Configure pre-commit hooks. My recommendation is to use this: https://pre-commit.com/. It should be as easy as `pip install pre-commit` or `brew install pre-commit` followed by `pre-commit install`. 
- Install `detect-secrets` (`pip install detect-secrets`) 
- Configure `pre-commit` to run `detect-secrets` by adding this file as `.pre-commit-config.yaml`:

```
repos:
-   repo: git@github.com:Yelp/detect-secrets
    rev: 0.10.3
    hooks:
    -   id: detect-secrets
        args: ['--baseline', '.secrets.baseline']
        exclude: .*/tests/.*
```

- Create a new branch: `git checkout -b foo`
- Edit the `README.md` and add `my password`
- Try to check it in: `git add README.md; git commit -m"committing a secret"`. Confirm that it fails. 
- Delete that branch: `git checkout -; git branch -d foo`

#### Have automated tests been added?

No. 

#### What are the relevant tickets?

https://3.basecamp.com/3098728/buckets/736178/todos/1350733790

#### How should this change be communicated to end users?

Let anyone who works on this repo know to configure this.

#### Next steps?

- See how well it works here and maybe start using it on other repos. 
- Probably don't want to scan that minified jQuery file.


#### Smells?

Not sure if we should commit that `.pre-commit-config.yaml` file. I didn't because someone may want to add to or use that file themselves to add other hooks. And if it were in the repo they couldn't do that.

#### Has the relevant documentation/wiki been updated?

Will after merge.
